### PR TITLE
Document Service Maturity

### PR DESCRIPTION
**Pull request by tobru**

### CODE REVIEW

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+TODO! To understand the personas referenced in this document, see xref:reference/glossary.adoc#_personas[Glossary - Personas].
+
+TODO! We currently maintain the service maturity assessment for each service on our product documentation site under https://products.docs.vshn.ch/products/appuio/managed/services_index.html[VSHN Application Catalog -> Services^].
+
+== Initial Release: Cluster Provisioning
+
+As a user of the service I can perform the following actions:
+
+Provisioning::
+I can provision a production-ready cloud Kubernetes environment on a major cloud service provider, such as Amazon EKS, Azure AKS, Google GKE, Exsoscale SKS (on request) and Swisscom ESC (on request).
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can provision a Cloud Kubernetes environment on our supported Cloud Service Provider.
```

I'd leave "production-ready" out here, it's unclear what this actually means. Maintaining the list of support Cloud Service Providers shouldn't be done here, instead it's part of products.docs.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+Scalability::
+I can easily scale my cluster as my needs change, without any downtime or disruption to my services.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Do we already want that in the provisioning iteration? Or should that be its own iteration ("Autoscaling")? Or should we clearly exclude autoscaling here to make it clear that we're not talking about that?

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+Customization::
+I can customize my cluster with the features and components I need to run my workloads.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Maybe say:

```suggestion
I can customize my cluster with a pre-defined set of features and components.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+Security::
+The service is built with security in mind and includes features such as encrypted communication, role-based access control (RBAC), and more to help me protect my data and workloads.
+
+Technical support::
+I can receive technical support from Kubernetes experts, who can help me resolve any issues or questions I may have, according to my SLA.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can receive technical support from Kubernetes experts, who can help me resolve any issues or questions I may have, according to my chosen xref:https://products.docs.vshn.ch/products/support_plans.html[Support Plan].
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+
+== Iteration: Cluster Maintainence
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
== Iteration: Cluster Maintenance
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+
+As a user of the service I can perform the following actions:
+
+Scheduling::
+I can schedule maintenance windows for my cloud Kubernetes environment, during which updates, patches, or other changes can be applied automatically.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can schedule maintenance windows for my Cloud Kubernetes environment, during which updates, patches, or other changes can be applied automatically.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+
+Notifications::
+I can receive notifications in advance of any scheduled maintenance, allowing me to plan accordingly and to minimize any disruption to my applications.
+
+Automated Updates::
+The service will automatically apply updates and patches to my cloud Kubernetes environment, ensuring that it is always running the latest and most secure version of the software.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
The service will automatically apply updates and patches to my Cloud Kubernetes environment, ensuring that it is always running the latest and most secure version of the software.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+
+Monitoring::
+I can monitor the status of my cloud Kubernetes environment during the maintenance window, to ensure that the maintenance is operating as expected.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can monitor the status of my Cloud Kubernetes environment during the maintenance window, to ensure that the maintenance is operating as expected.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+
+
+== Iteration: Cluster Upgrade
+
+As a user of the service I can perform the following actions:
+
+Upgrade::
+I can upgrade my cloud Kubernetes clusters minor version.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can upgrade my Cloud Kubernetes clusters minor version.
```

Maybe rephrase into "I can schedule an automated upgrade of my Cloud Kubernetes cluster to the next minor version".

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+As an engineer of the service I can perform the following actions:
+
+Upgrade::
+I can upgrade my cloud Kubernetes clusters minor version.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Regarding the above suggestion:

```suggestion
I can trigger an automated upgrade of my Cloud Kubernetes cluster to the next minor version
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+
+
+== Iteration: Backup and Restore
+
+
+As a user of the service I can perform the following actions:
+
+Data backup::
+I can create backups of my cloud Kubernetes environment, including the configuration, state, and data of my applications.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can rely on automated backups of my Cloud Kubernetes cluster configuration and state
```

In APPUiO Managed, we don't care about the data of the end-users applications running on the cluster. We only care about the cluster itself and provide means for the end-user to do application backup, aka we provide a well-configured K8up.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+
+Testing::
+I can test restores in a development environment before performing a full restore, to ensure that my data and applications are fully recoverable.
+
+
+== Iteration: Monitoring and Alerting
+
+As a user of the service I can perform the following actions:
+
+* I can have a monitoring service that will help me keep track of the health and performance of my cloud Kubernetes environment.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
* I have a monitoring service that will help me keep track of the health and performance of my cloud Kubernetes environment.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 13]
```diff
@@ -0,0 +1,294 @@
...
+* I can enable alerting on pre-defined SLIs to VSHN so that VSHN can resolve incidents with priority.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
* I have alerting on pre-defined SLIs to VSHN so that VSHN can resolve incidents with priority.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 14]
```diff
@@ -0,0 +1,294 @@
...
+I can view and access my running costs of the cloud Kubernetes environment
+
+As an engineer of the service I can perform the following actions:
+
+SLA::
+When a request for changing the SLA of a Kubernetes environment is approved, I can change the SLA.
+
+SLA::
+I can view the SLA of a Kubernetes environment per customer and per cluster
+
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc:90]:
```suggestion
Automated Billing::
The service instances are automatically billed according to the defined price model.
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 15]
```diff
@@ -0,0 +1,294 @@
...
+As a user of the service I can perform the following actions:
+
+Cluster Logs::
+I can access the logs of the Kubernetes control plane via a user-friendly interface
+
+Application Logs::
+I can access the logs of my applications via a user-friendly interface
+
+Log retention::
+I can configure the retention of the logs in the cloud provider's logging stack
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can configure the retention of the logs in the provided logging stack
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 16]
```diff
@@ -0,0 +1,294 @@
...
+I can define rules to route incoming traffic to specific services based on the URL path, host, or other criteria.
+
+Load balancing::
+I can configure the Ingress Controller to load balance incoming traffic across multiple replicas of my services to improve reliability and performance.
+
+SSL/TLS termination::
+I can configure the Ingress Controller to terminate SSL/TLS connections and handle encryption/decryption of incoming traffic, making it easier to secure my services.
+
+Custom configuration::
+I can customize the behavior of the Ingress Controller by providing custom configuration files, such as custom error pages or redirect rules.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Maybe combine this into:

```suggestion
Ingress Controller::
I have a modern Ingress controller available to expose my applications using the `Ingress` object.
```

(Or even "using the `Ingress` and more modern Gateway API").

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 17]
```diff
@@ -0,0 +1,294 @@
...
+As an engineer of the service I can perform the following actions:
+
+Monitoring and logging::
+I can monitor and log incoming traffic through the Ingress Controller, helping me to track potential problems of the Ingress Controller
+
+Custom configuration::
+I can customize the behavior of the Ingress Controller by providing custom configuration files, such as global defaults for certain timeouts
+
+
+== Iteration: Cert Manager
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
== Iteration: Automated Certificate Handling
```

Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: This section reads like a feature list of what Cert Manager does. I'm uncertain if this provides any value to the user. It could probably make sense to have a higher level list stating that we provide a pre-configured "Cert Manager" and other components?

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 18]
```diff
@@ -0,0 +1,294 @@
...
+Namespace-level RBAC::
+I can enforce RBAC at the namespace level, allowing me to define different access policies for different parts of their cluster.
+
+
+== Iteration: Container Network Interface (CNI)
+
+As an engineer of the service I can perform the following actions:
+
+Network plugin selection::
+I can choose from a variety of VSHN supported CNI plugins
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Do we really want that? I'd stick with one default CNI.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 19]
```diff
@@ -0,0 +1,294 @@
...
+
+Network customization::
+I can customize the network configuration to meet the specific needs of the user's workloads and applications, such as specifying network segmentation, IP address ranges, and other network-level attributes.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: I propose to have a clearly defined list of things which can be customized.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 20]
```diff
@@ -0,0 +1,294 @@
...
+Network performance::
+I can use CNI plugins to optimize network performance for the user's workloads and applications
+
+
+== Iteration: LoadBalancer
+
+As an engineer of the service I can perform the following actions:
+
+Load balancing setup::
+I can set up load balancing for the applications and services, using the Ingress-Controller. This allows me to define how incoming traffic to the applications should be routed and balanced, ensuring that the users have a seamless and reliable experience.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Are we talking about the Service type `LoadBalancer` in this section? If you, what role does the Ingress controller play here?

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 1 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Oh, and we have to bring in the "self-service" aspect here somehow.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 21]
```diff
@@ -0,0 +1,294 @@
...
+Upgrade::
+I can upgrade my cloud Kubernetes clusters minor version.
+
+
+== Iteration: Backup and Restore
+
+
+As a user of the service I can perform the following actions:
+
+Data backup::
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
Cluster configuration backup::
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 11 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Add a part "Application data backup::"

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 22]
```diff
@@ -0,0 +1,294 @@
...
+I can bind roles to users or groups of users, effectively granting them the specified permissions.
+
+Role management::
+I can manage my roles and role bindings using Kubernetes APIs, allowing me to dynamically control access to their cluster and its resources.
+
+Cluster-level RBAC::
+I can enforce Cluster RBAC at the cluster level, ensuring that all users and groups have the appropriate permissions and access to their cluster and its resources.
+
+Namespace-level RBAC::
+I can enforce RBAC at the namespace level, allowing me to define different access policies for different parts of their cluster.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Remove that section, it currently doesn't add any value.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 18 above)
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
I can choose from a pre-defined list of VSHN supported CNI plugins
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 23]
```diff
@@ -0,0 +1,294 @@
...
+Log retention::
+I can configure the retention of the logs in the cloud provider's logging stack
+
+As an engineer of the service I can perform the following actions:
+
+Cluster Logs::
+I can access the logs of the Kubernetes control plane via a user-friendly interface
+
+
+== Iteration: Ingress Controller
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]:
```suggestion
== Iteration: Service Exposure
```

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 1 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: For us (colin & Hadar) it is clear it is a self-service. because it is referring to the End User capabilities.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 2 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Do we want to add more info regarding the autoscalling here in the service maturity or should we leave it out and add it only to the product description? (as we also said we would like to shorten the service maturity list..)

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 11 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: We thought about it again and we think that there is no need to add the "Application data backup" here as this is a service that will be provided by AppCat, as the ownership of K8up is at AppCat.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 12 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: As discussed in the meeting we had, we agreed to keep its as it is and not change "can" to "have". so we leave it as it is right now.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 13 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: As discussed in the meeting we had, we agreed to keep its as it is and not change "can" to "have". so we leave it as it is right now.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 15 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Was not added at the end, as we deleted all information about the logs, except for the Cluster Logs.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 24]
```diff
@@ -0,0 +1,294 @@
...
+
+Custom configuration::
+I can customize the behavior of the Ingress Controller by providing custom configuration files, such as custom error pages or redirect rules.
+
+Monitoring and logging::
+I can monitor and log incoming traffic through the Ingress Controller, helping me to track traffic patterns and identify issues.
+
+Monitoring::
+I can monitor the expiration of my certificates
+
```
Comment (shera55) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: We suggest to delete it from this Iteration, as every component can be monitored and we can mention it at the Monitoring Iteration.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 25]
```diff
@@ -0,0 +1,294 @@
...
+As an engineer of the service I can perform the following actions:
+
+Monitoring and logging::
+I can monitor and log incoming traffic through the Ingress Controller, helping me to track potential problems of the Ingress Controller
```
Comment (shera55) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: We suggest to delete it from this Iteration, as every component can be monitored and we can mention it at the Monitoring Iteration.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 26]
```diff
@@ -0,0 +1,294 @@
...
+
+Custom configuration::
+I can customize the behavior of the Ingress Controller by providing custom configuration files, such as global defaults for certain timeouts
+
```
Comment (shera55) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: We suggest to delete it from this Iteration, as every component can be customized and since we suggested to remove also the monitoring part, we suggest to delete the entire section of "as a VSHNeer Engineer".

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 1 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Always be explicit in such documents, don't make assumption. For me, it wasn't clear, and therefore I strongly suggest being precise. That's why I propose to bring the self-service aspect into it, as this is one of the most important pillars of the product.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 2 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: As autoscaling is an important aspect of the product, I propose we add a section especially for that topic.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 11 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Maybe that's true, but I'm not sure about it, because I think providing some application supporting functionality is part of the platform. Let's discuss that.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 24 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: I agree that one doesn't really fit here. It would probably be a better fit for "User Workload Monitoring" which I'm unsure if this is yet another iteration or an additional component / add-on with its own features and iterations.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 25 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Agreed :+1:

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 26 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Yes, I agree, I'd remove that section as well. We can always iterate over it as we learn.

File: docs/modules/ROOT/pages/references/glossary.adoc [hunk 27]
```diff
@@ -1,7 +1,72 @@
...
+Architecture refers to the deployment strategy of a service instance in relation to high availability.
+
+[[standalone]]Standalone:::
+A single node instance without any high availability.
+
+[[replicated]]Replicated:::
+The service offers data replication to one or more other instances.
+
+[[clustered]]Clustered:::
+A fully clustered version of the service, usually offering active / active replication with fully automated cluster handling.
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/glossary.adoc]: Is there a reason why you copied the full glossary from AppCat? IMHO the appcat glossary doesn't fully apply to APPUiO Managed Kubernetes Services. So I'd opt to remove that again from this document here.

File: docs/modules/ROOT/pages/references/glossary.adoc [hunk 28]
```diff
@@ -1,7 +1,72 @@
...
+
+[[service]]Service::
+A service that can be ordered.
+It is automatically provisioned, can be consumed by a service consumer and generally adheres to our quality requirements.
+
+[[service-consumer]]Service Consumer::
+An application that connects to a service instance as a client.
+
+[[service-order]]Service Order::
+The act of deploying a Kubernetes resource that triggers the provisioning of a new service instance.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/glossary.adoc]: Is _Service Order_ a noun or a verb? To me, it feels more like a noun.

```suggestion
A Kubernetes resource defining a service instance and governing the provisioning of a new service instance.
```

File: docs/modules/ROOT/pages/references/glossary.adoc [hunk 29]
```diff
@@ -1,7 +1,72 @@
...
+
+== Personas
+
+[[End-user]]End User::
+A human that interacts with the service (uses it).
+
+[[Technical-VSHNeer]]Technical VSHNeer::
+A VSHNeer doing system administration, solution architecture and other technically focused jobs.
+
+== How-to guides
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/glossary.adoc]: Looks misplaced. I would not expect a list of how-tos to be listed on a glossary page.

File: docs/modules/ROOT/pages/references/glossary.adoc [hunk 30]
```diff
@@ -1,7 +1,72 @@
...
+The act of deploying a Kubernetes resource that triggers the provisioning of a new service instance.
+
+[[service-instance]]Service Instance::
+A single isolated occurrence of a service.
+
+[[cloud-instance]]Cloud Instance:::
+An instance of a cloud service offering by a cloud service provider.
+
+[[converged-service-instance]]Converged Service Instance:::
+An instance of a converged service blueprint.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/glossary.adoc]: And what is a _converged service blueprint_?

File: docs/modules/ROOT/pages/references/glossary.adoc [hunk 31]
```diff
@@ -1,7 +1,72 @@
 = Glossary
 
-https://azure.microsoft.com/de-de/services/kubernetes-service/[AKS]:: Azure Kubernetes Service (AKS)
+////
+You can link these terms using xrefs and their IDs, for example:
+  xref:reference/glossary.adoc#service-instance[Service Instance]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/glossary.adoc]: You can also use `<<service-instance>>` to cross-link on the same page. I suggest doing so for all glossary terms referring to another term.

File: docs/modules/ROOT/pages/references/glossary.adoc (hunk 27 above)
Comment (corvus-ch) [reply docs/modules/ROOT/pages/references/glossary.adoc]: Yep, that was confusing. Up until now, I had the impression to read an AppCat related documentation.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
This look like to be a copy from the AppCat documentation with no or just a few changes. This is very confusing. Please make this more specific to Cloud Kubernetes.

File: docs/modules/ROOT/pages/references/glossary.adoc (hunk 27 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/glossary.adoc]: I copied part of AppCat glossary cause I thought we want to align it to APPUiO Managed Kubernetes Services down the road, once we add more information. Sure, I will remove it.

File: docs/modules/ROOT/pages/references/glossary.adoc (hunk 29 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/glossary.adoc]: This list was already on the glossary page; I just added the header and edited it a bit.. I assume @tobru added it to the glossary page, maybe he can share his opinion as well?

Comment (tobru) [reply docs/modules/ROOT/pages/references/glossary.adoc]: The list was already there, see https://kb.vshn.ch/cloud-kubernetes/references/glossary.html

But the title "How-to guides" was introduced in this (104ed609617c67d4d1789e6007df95a1d060ca9f) commit, which is a mistake.

File: docs/modules/ROOT/pages/references/glossary.adoc (hunk 27 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/glossary.adoc]: Yes, better get rid of this part again, this confuses too much and doesn't apply for this product.

**Comment by tobru** [comment]
@corvus-ch @shera55 I reverted the last changes which messed up with the content, it's now once again at the stage where it's ready to be reviewed.

**Comment by shera55** [comment]
@tobru I reviewed both files and it looks good. Nothing to add from my side. Thank you.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 32]
```diff
@@ -0,0 +1,142 @@
...
+* connect to the Cloud Kubernetes API for using it.
+* customize my cluster with a pre-defined set of features and components.
+** scale my cluster horizontally and vertically, as my needs change.
+** choose the https://products.docs.vshn.ch/products/service_levels.html[service level^] of the instance.
+* delete the Cloud Kubernetes environment again.
+
+As a *Technical VSHNeer* I can:
+
+* see and configure the https://products.docs.vshn.ch/products/service_levels.html[service level^] of the instance.
+* automatically bill the service to the end-user according to the specified price model.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/service-maturity.adoc:32]: Maybe that is a role on its own? A _Technical VSHNeer_ actually does not necessarily care about billing (at least not directly). It is actually VSHN as a whole that needs this.

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 33]
```diff
@@ -0,0 +1,142 @@
...
+
+== Iteration: Cluster Backup and Restore
+
+As an *End-User* I can:
+
+* rely on automated backups of my Cloud Kubernetes environment configuration and cluster state.
+* schedule backups cluster state backups to run automatically, at a time and frequency that suits my needs.
+* configure data retention policies to ensure that backups are kept for the desired length of time.
+* quickly and easily restore my Cloud Kubernetes environment to a previous state, including all cluster configuration.
+* test restores in a test environment before performing a full restore, to ensure that the backup is fully restorable.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: That is a real tough one. It probably needs clarification of scope. This can read as _restore a cluster along with everything deployed on it_. Just a few points where this might go wrong:

* PVs can only be bound to one cluster or might not be available anymore
* Deployments might reference external resources (S3, Cloud databases, APIs). So the copy of an application might start to fight about state with the actual production one)
* And probably a lot more

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 34]
```diff
@@ -0,0 +1,142 @@
...
+* enable alerting on pre-defined SLIs to VSHN so that VSHN can resolve incidents with priority.
+* see SLO reports so that I can get an overview of how well my Cloud Kubernetes environment performs.
+* access operations runbooks so that I can resolve upcoming alerts.
+* access a metrics dashboard to get an overview of the Cloud Kubernetes environment and health.
+
+== Iteration: End-User Metrics and Alerting
+
+As an *End-User* I can:
+
+* enable a service to automatically collect and aggregate metrics from my Cloud Kubernetes environment and applications, providing a unified view of all metrics.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/service-maturity.adoc:82]: This looks like a duplication of line 69. But I guess the difference is one cluster vs several clusters. For the latter, I wonder if this is a feature of Cloud Kubernetes or a product on its own that can be used by everything VSHN provides?

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 35]
```diff
@@ -0,0 +1,142 @@
...
+
+As an *End-User* I can:
+
+* enable collection of historic logs of my applications running on the cluster.
+* access historic logs of my application running on the cluster.
+* configure retention time of historic logs.
+
+As a *Technical VSHNeer* I can:
+
+* access the logs of the Kubernetes control plane via a user-friendly interface.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/service-maturity.adoc]: Nice _user-friendly_ is only relevant for VSHNeers but is not mentioned for end-users 🤣

File: docs/modules/ROOT/pages/references/service-maturity.adoc [hunk 36]
```diff
@@ -0,0 +1,142 @@
...
+
+As an *End-User* and *Technical VSHNeer* I can:
+
+* log in to my cluster through a user-friendly interface
+
+== Iteration: Networking
+
+As a *Technical VSHNeer* I can:
+
+* choose from a pre-defined list of CNI plugins.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/service-maturity.adoc:139]: Does the decision to only use Cilium only apply to OpenShift?

**Review by corvus-ch — COMMENTED** [review]
I have left some remarks on things that caught my eye.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 32 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc:32]: You're right. Coming up with another Persona is something I'd like to invest a bit more time and for the time being, leave it as it is. Once we fine-tuned to personas, we can adapt it here as well.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 33 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: I strongly agree. I'd even say: Let's remove that one for now, it could make this iteration into a state where it isn't possible to get it done without a huge amount of effort.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 34 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc:82]: The differences between this line and line 69 is the context: Line 69 is in the "Cluster-Level Metrics" context and that one in the "End-User Metrics" context. Comparing that to OpenShift would mean: Here we're talking about the "User Workload Monitoring" feature and in line 69 about the standard cluster monitoring. Do you see anything which we could rephrase so that this gets more clear?

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 35 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: That wasn't the intention, and slipped through my fingers. I'd remove "user-friendly" here, as this is not a well-defined wording. What does "user-friendly" actually mean? Define "user-friendly"...

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 36 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc:139]: The idea was to not specify the exact CNI in this document, but in another "pre-defined list of CNI plugins", which then will state that Cilium is the default.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 34 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc:82]: I would suggest to maybe change the name of the iteration to "Applications Metrics and Alerting". In addition to what @corvus-ch mentioned, and as mentioned also before, I think that this could also be a product on its own.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 35 above)
Comment (shera55) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Maybe instead of user-friendly interface we can write "access the logs of the Kubernetes control plane via a UI dashboard" or "via an intereactive dashboard". What do you think about that? 

And on that note, we wrote it a bit differently for the End-User ("access historic logs...") - maybe we can also edit this sentence for the End-User? i.e to add at the end of this sentence the "via a UI dashboard" or "via an intereactive dashboard".

Another thing I think we should consider: do we want to have a feature for the End-User application mentioned here as well or if we decide it will be a standalone product, then maybe we should remove it from this iteration?

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 33 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: I removed that line now.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 34 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc:82]: Yes, I agree, I renamed the iteration. For now, I'd like to keep it here as this is an important aspect for an end-user of the product, and I'm pretty sure we have to offer that by default not as an add-on.

File: docs/modules/ROOT/pages/references/service-maturity.adoc (hunk 35 above)
Comment (tobru) [reply docs/modules/ROOT/pages/references/service-maturity.adoc]: Good input, I rephrased it accordingly.

**Review by corvus-ch — APPROVED** [review]

**Review by shera55 — APPROVED** [review]
